pkg/governance-scripts/test/TimelockAuthorizerTransitionMigrator.test.ts

```diff
@@ -151,6 +153,107 @@ describe('TimelockAuthorizerTransitionMigrator', () => {
         });
       });
+
+      describe('delayed permissions', () => {
+        const delay = DAY;
+        let grantActionId: string;
+        let delayedRoleData: RoleData;
+
+        async function setDelay(actionId: string, delay: number) {
+          const receipt = await newAuthorizer.connect(root).scheduleDelayChange(actionId, delay, []);
+          const event = expectEvent.inReceipt(await receipt.wait(), 'ExecutionScheduled');
+          const scheduledExecutionId = event.args.scheduledExecutionId;
+          await advanceToTimestamp((await newAuthorizer.getScheduledExecution(scheduledExecutionId)).executableAt);
+          await newAuthorizer.execute(scheduledExecutionId);
+        }
+
+        sharedBeforeEach('set delay', async () => {
+          const setAuthorizerAction = await actionId(vault, 'setAuthorizer');
+          await setDelay(setAuthorizerAction, delay * 2);
+
+          delayedRoleData = rolesData[0];
+          grantActionId = await newAuthorizer.getGrantPermissionActionId(delayedRoleData.role);
+          await setDelay(grantActionId, delay);
+        });
+
+        context('when executing scheduled permissions before migrating', () => {
+          it('reverts', async () => {
+            await expect(transitionMigrator.executeScheduledPermissions()).to.be.revertedWith('MIGRATION_INCOMPLETE');
+          });
+        });
+
+        context('when migrating before executing scheduled permissions', () => {
+          let receipt: ContractReceipt;
+          let scheduledExecutionId: BigNumber;
+
+          sharedBeforeEach('migrate permissions and store scheduled permission ID', async () => {
+            const tx = await transitionMigrator.migratePermissions();
+            receipt = await tx.wait();
+            scheduledExecutionId = await transitionMigrator.scheduledPermissionIds(0);
+          });
+
+          it('migrates all non-delayed permissions', async () => {
+            for (const roleData of rolesData) {
+              if (roleData === delayedRoleData) {
+                expect(await newAuthorizer.canPerform(roleData.role, roleData.grantee, roleData.target)).to.be.false;
+              } else {
+                expect(await newAuthorizer.canPerform(roleData.role, roleData.grantee, roleData.target)).to.be.true;
+              }
+            }
+          });
+
+          it('stores scheduled permission ID', async () => {
+            expectEvent.inIndirectReceipt(
+              receipt,
+              newAuthorizer.interface,
+              'ExecutionScheduled',
+              {
+                actionId: grantActionId,
+                scheduledExecutionId,
+              },
+              newAuthorizer.address,
+              1
+            );
+
+            await expect(transitionMigrator.scheduledPermissionIds(1)).to.be.reverted;
+          });
+
+          it('can grant a scheduled permission after the delay passes', async () => {
```

Comment:
Maybe have 2 permissions with different delays? And test that it grants the earliest and keeps the second one pending until the longer delay expires?

Reply:
Done in https://github.com/balancer-labs/balancer-v2-monorepo/pull/2184/commits/af8a8ba369d0fe67ea729c7cf359dc4d025f7353.